testing/e2e/suites/defaultdomain.go

```diff
@@ -240,6 +241,53 @@ func defaultDomainTests(in infra.Provisioned) []test {
 					return fmt.Errorf("parsing and verifying TLS certificate: %w", err)
 				}
 
+				lgr.Info("Validating Rotation Story")
+				certPEM, keyPEM, err := manifests.GenerateSelfSignedCert()
+				if err != nil {
+					return fmt.Errorf("generating self-signed cert: %w", err)
+				}
+
+				dds := manifests.CreateDefaultDomainSecret(certPEM, keyPEM)
+				if err := util.Upsert(ctx, cl, dds); err != nil {
+					return fmt.Errorf("upserting DefaultDomainSecret: %w", err)
+				}
+
+				rotatedSecret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretTarget,
+						Namespace: namespace.GetName(),
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Secret",
+						APIVersion: corev1.SchemeGroupVersion.String(),
+					},
+				}
+				if err := cl.Get(ctx, client.ObjectKeyFromObject(rotatedSecret), rotatedSecret); err != nil {
+					return fmt.Errorf("getting Secret %s/%s: %w", rotatedSecret.Namespace, rotatedSecret.Name, err)
+				}
+
+				rotatedCert, ok := rotatedSecret.Data["tls.crt"]
+				if !ok {
+					return fmt.Errorf("Secret %s/%s does not contain tls.crt data", rotatedSecret.Namespace, rotatedSecret.Name)
+				} else if !bytes.Equal(rotatedCert, certPEM) {
```

Comment:
don't use else if. just use if

we return early if it doesn't work and else ifs are just harder to read.

This is golang convention